src/main/ruby/core/enumerable.rb

```diff
@@ -112,6 +112,7 @@ def count(item = undefined)
     seq = 0
     if !undefined.equal?(item)
       each { |o| seq += 1 if item == o }
+      warn 'warning: given block not used' if block_given?
```

Comment:
I tried to match the MRI behavior which only warns after iteration stops. I did expect it to warn up front.

```ruby
[sleep(1), sleep(1)].count(nil) { }
# after two seconds:
#!> warning: given block not used
```